CMakeLists.txt

```diff
@@ -355,6 +353,8 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -fopenmp=libomp -Wall -Wextra -pedantic")
 endif()
+
+find_package(FILESYSTEM REQUIRED)
 target_link_libraries(SeisSol-lib PUBLIC ${FILESYSTEM_LIBRARIES})
```

Comment:
Is it possible to create a target for filesystem?

Reply:
What do you think about this one?
```
if (DEFINED FILESYSTEM_LIBRARIES)
    add_library(std::filesystem INTERFACE IMPORTED)
    set_property(TARGET std::filesystem APPEND PROPERTY INTERFACE_COMPILE_FEATURES cxx_std_17)

    if (FILESYSTEM_LIBRARIES)
        set_property(TARGET std::filesystem APPEND PROPERTY INTERFACE_LINK_LIBRARIES ${FILESYSTEM_LIBRARIES})
    endif()
endif()
```

Reply:
What do you think about this one?

```
if (DEFINED FILESYSTEM_LIBRARIES)
    add_library(std::filesystem INTERFACE IMPORTED)
    set_property(TARGET std::filesystem APPEND PROPERTY INTERFACE_COMPILE_FEATURES cxx_std_17)

    if (FILESYSTEM_LIBRARIES)
        set_property(TARGET std::filesystem APPEND PROPERTY INTERFACE_LINK_LIBRARIES ${FILESYSTEM_LIBRARIES})
    endif()
endif()
```

[source](https://storage.bsc.es/gitlab/hpc/gekkofs/-/blob/c9ce879500713b0fb91cd284844af6bb1fd510c7/CMake/FindFilesystem.cmake)

Reply:
looks good

Reply:
Ok. I tested with gcc@8, gcc@9 and clang@11 with  `std::filesystem` target and it worked for me